testsuite/gnat2goto/tests/type_modular_non_power_of_two_decl/modular_type.adb

```diff
@@ -0,0 +1,9 @@
+procedure Modular_Type is
+  type Less_Than_Ten is mod 10;
+  procedure Assert_Val (X : Less_Than_Ten) is
+  begin
+    pragma Assert (X = 5);
```

Comment:
So I guess you won't be able to compare it with anything other than mod-10 type?

Reply:
No, that'd be rejected by the frontend.

Reply:
Well on the CBMC side the type is bitvector (but with named sub) so it cannot really be compared with normal bitvectors. But, as you say, that's probably intended.